teos/watcher.py

```diff
@@ -673,11 +673,44 @@ def get_user_info(self, user_id):
             user_id (:obj:`str`): the id of the requested user.
 
         Returns:
-            :obj:`UserInfo <teos.gatekeeper.UserInfo> or :obj:`None`: The user data if found. :obj:`None` if not found, or
-            the ``user_id`` is invalid.
+            :obj:`UserInfo <teos.gatekeeper.UserInfo> or :obj:`None`: The user data if found. :obj:`None` if not found,
+            or the ``user_id`` is invalid.
         """
         return self.gatekeeper.registered_users.get(user_id)
 
+    def get_subscription_info(self, signature):
+        """
+        Gets information about a user's subscription.
+
+        Args:
+            signature (:obj:`str`): the signature of the request by the user.
+
+        Returns:
+            :obj:`UserInfo <teos.gatekeeper.UserInfo> or :obj:`None`: The user's subscription data if found. :obj:`None`
+            if not found, or the ``user_id`` is invalid.
+
+        Raises:
+            :obj:`SubscriptionExpired`: If the user subscription has expired.
+        """
+
+        message = "get subscription info".encode("utf-8")
+        user_id = self.gatekeeper.authenticate_user(message, signature)
+        if self.gatekeeper.has_subscription_expired(user_id):
+            raise SubscriptionExpired(
+                f"Your subscription expired at block {self.gatekeeper.registered_users[user_id].subscription_expiry}"
+            )
+
+        subscription_info = self.gatekeeper.registered_users.get(user_id)
+
+        locators = []
+
+        for appt_uuid in subscription_info.appointments:
```

Comment:
There's no need to go to the DB for this, you can check `self.appointments.get(uuid)` and get the locator (if it is on the `Watcher`). If the data is not there, it must be on the `Responder`, so `self.responder.trackers.get(uuid)`.

For both the field `locator` should be set.